packages/stockflux-launcher/src/App.js

```diff
@@ -1,69 +1,112 @@
+import { ScreenEdge, useDockWindow, useOptions } from 'openfin-react-hooks';
+import React, { useEffect, useState } from 'react';
+import Components from 'stockflux-components';
 import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
-import {
-  FaChevronUp,
-  FaChevronLeft,
-  FaChevronRight,
-  FaRegHandRock
-} from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
+import './App.css';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
+import getUndockedPosition from './helpers/getUndockedPosition';
+import { ReactComponent as LeftIcon } from './icons/left.svg';
+import { ReactComponent as RightIcon } from './icons/right.svg';
+import { ReactComponent as TopIcon } from './icons/top.svg';
+import Titlebar from './titlebar/Titlebar';
 import ToolBar from './toolbar/ToolBar';
-import CloseButton from './toolbar/CloseButton';
-import './App.css';
-
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
 
 export default () => {
+  const [options] = useOptions();
+  const defaultHeight = options ? options.defaultHeight : 75;
+  const defaultWidth = options ? options.defaultWidth : 50;
+  const isDockable = options ? options.customData.isDockable : false;
+  const [leftPosition, setLeft] = useState(0);
+  const [topPosition, setTop] = useState(0);
+  const [isHorizontal, setHorizontal] = useState(true);
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
     OpenfinApiHelpers.getCurrentWindowSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    {
+      dockedWidth: isHorizontal ? defaultWidth : 80,
+      dockedHeight: defaultHeight
+    },
+    {
+      undockPosition: { left: leftPosition, top: topPosition },
+      undockSize: { width: 724, height: 88 }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  const iconStyle = isHorizontal ? 'icon horizontal' : 'icon vertical';
+  const toolbarStyle = isHorizontal ? 'toolbar' : 'toolbar t-vertical';
+
+  /* Hook undock if initialDocked is false on start */
+  useEffect(() => {
+    if (
+      options &&
+      options.customData &&
+      options.customData.initialDocked === false
+    ) {
+      windowActions.dockNone();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
+
+  /* Make sure window shrinks when dragged from top */
   useEffect(() => {
-    prevEdgeRef.current = edge;
-  });
-  const prevEdge = prevEdgeRef.current;
+    if (edge === ScreenEdge.NONE) {
+      getUndockedPosition().then(position => {
+        setLeft(position.left);
+        setTop(position.top);
+
+        windowActions.dockNone();
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [edge]);
 
-  const edgeToBeChecked = edge === ScreenEdge.NONE ? prevEdge : edge;
+  useEffect(() => {
+    setHorizontal(
+      edge === ScreenEdge.TOP || edge === ScreenEdge.NONE ? true : false
+    );
+  }, [edge]);
```

Comment:
Again, I think it's a bit overkill having this in `useEffect`, it could just be a helper function that gets called only when we need to know if the launcher is horizontal or not rather than rerendering everytime `edge` form dependency array changes.